tests/crypto/test_keyring.py

```diff
@@ -86,35 +86,6 @@ def check_context(self, _, expected):
             getattr(LoggingContext.current_context(), "request", None), expected
         )
 
-    def test_wait_for_previous_lookups(self):
```

Comment:
Should there not be a new test case to ensure that we do actually take out a lock? Or is there already such a test?

Reply:
test_verify_json_objects_for_server_awaits_previous_requests checks that we don't end up with two lookups at once.